app/policies/test_result_policy.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# Policies for accessing TestResults
+class TestResultPolicy < ApplicationPolicy
+  def index?
+    Pundit.policy(user, record.profile)
```

Comment:
:thinking: if the profile is deleted but not the report, this breaks

```suggestion
    Pundit.policy(user, record.profile) || Pundit.policy(user, record.host)
```